### PR TITLE
Move light position

### PR DESCRIPTION
**Pull request by kerautret**
add a new possibility to move light source position (an x-axis (resp. y-axis) mouse move changes the azimuth (resp. inclination angle) of the light source).

### CODE REVIEW

**Comment by kerautret** [comment]
The light source direction is also displayed during the light source move.

File: ChangeLog.md [hunk 1]
```diff
@@ -13,6 +13,11 @@
  - Minor improvements of default settings in Viewer3D. (David
    Coeurjolly, [#1066](https://github.com/DGtal-team/DGtal/pull/1066))
 
+ - New possibility to move the light source according to the mouse move in
```
Comment (dcoeurjo) [inline ChangeLog.md]: ... source using the mouse (with SHIFT key modifier) in Viewer3D

File: src/DGtal/io/viewers/Viewer3D.h [hunk 2]
```diff
@@ -1267,7 +1293,11 @@ namespace DGtal
     float myGLScaleFactorX;
     float myGLScaleFactorY;
     float myGLScaleFactorZ;
-
+    // Used to apply interactive light rotation
+    float myLigthRotationStep;
+    int myRefMouseXPos;
+    int myRefMouseYPos;
```
Comment (dcoeurjo) [inline src/DGtal/io/viewers/Viewer3D.h]: doxygen doc of the members ?

**Comment by dcoeurjo** [comment]
Nice feature ;) i'll test it

File: ChangeLog.md (hunk 1 above)
Comment (kerautret) [inline ChangeLog.md]: better thanks ;)

File: src/DGtal/io/viewers/Viewer3D.h (hunk 2 above)
Comment (kerautret) [inline src/DGtal/io/viewers/Viewer3D.h]: ok

**Comment by kerautret** [comment]
done ;)

**Comment by dcoeurjo** [comment]
x-y seems to be ok but I don't understand the "z"
BTW add the key binding (shift+CMD on mac..) in the changelog
and add something to say that the light always looks at (0,0,0)

**Comment by kerautret** [comment]
the position z of the light should change when you move the mouse according an y move (upper or lower). 
ok for the key

**Comment by dcoeurjo** [comment]
euh.. I don"t get it.. Are x (resp. y) moves increasing the phi angle (resp. theta) in a Euler representation ?

**Comment by kerautret** [comment]
yes exactly

**Comment by kerautret** [comment]
I fix a pb of direction update from the draw(), now it should be finer. The light source direction is always defined in the main system coords, and don't move with the camera. I think that this choice is better to simplify the user interaction with the definition of move.
Hope that it is also fine for you?

**Comment by dcoeurjo** [comment]
perfect!! thanks a lot

**Comment by dcoeurjo** [comment]
Would it be possible to add a line in the documentation ?

**Comment by kerautret** [comment]
it in the help function but you mean in the main section ok I add it

**Comment by dcoeurjo** [comment]
I meant in the module documentation

**Comment by kerautret** [comment]
done

**Comment by dcoeurjo** [comment]
great
merging

**Pull request merged**